src/routes/+page.svelte

```diff
@@ -1,79 +1,82 @@
+    if (index === 0) return { card: "full", button: "primary", imageLoading: "eager" };
+    else if (index < 3) return { card: "half", button: "secondary", imageLoading: "eager" };
+    else return { card: "third", button: "outline", imageLoading: "lazy" };
+  };
 </script>
 
-<main id="main-content">
-  <Image
-    class="main-image"
-    alt=""
-    loading="eager"
-    src={landingImage}
-    sources={[
-      { type: "image/avif", srcset: [landingImageAvif, [landingImageAvifMobile, 412]] },
-      { type: "image/webp", srcset: [landingImageWebP, [landingImageWebpMobile, 412]] },
-      { type: "image/jpeg", srcset: [landingImage, [landingImageMobile, 412]] },
-    ]}
-    blurhash={landingImageBlurhash}
-    width={landingImageWidth}
-    height={landingImageHeight}
-    sizeType="full-bleed"
-    mean={landingImageMean}
-    preserveAspectRatio={false}
-    style="object-position: 0 25%"
-  />
-  <section class="grid-container homepage-greeting maxw-tablet">
-    <ContentfulRichText document={data.title} />
-    <div class="grid-row">
-      <div class="grid-col-4 construction-sign-container">
-        <Image
-          src={underConstructionGif}
-          sources={[
-            { type: "image/avif", srcset: [underConstructionAvif] },
-            { type: "image/gif", srcset: [underConstructionGif] },
-          ]}
-          alt="Under construction sign swinging in the wind"
-          width={underConstructionWidth}
-          height={underConstructionHeight}
-          fit={false}
-        />
-      </div>
-      <div class="grid-col-8 padding-left-4">
-        <ContentfulRichText document={data.body} />
-      </div>
-    </div>
-  </section>
-  <!-- TODO: Uncomment when working on homepage, provide data from Contentful. -->
-  <!--
-  <div class="grid-container margin-top-2">
-    <VideoCard url="https://www.youtube.com/watch?v=3HRGzFjA_cU" variation="hero" />
-  </div>
-  -->
+<main id="main-content" class="grid-container usa-prose margin-top-2">
+  {#if heroVideo?.videoUrl}
+    {@const { videoUrl, videoTitle, videoSubhead } = heroVideo}
+    <VideoCard
+      url={videoUrl}
+      customTitle={videoTitle}
+      customDescription={videoSubhead}
+      variation="hero"
+    />
+  {/if}
+  {#if featuredServices}
+    <ul class="service-group-list">
+      {#each featuredServices as item, index (item?.pageMetadata?.sys.id)}
+        {@const { card: cardSize, button: buttonVariant, imageLoading } = getCardSettings(index)}
+        <!-- TODO: Can't conditionally render a named slot, but ideally we only declare Card once here. -->
+        {#if item?.heroImage?.imageSource?.url}
+          <Card class={`usa-card--${cardSize}`}>
+            <h3 class="usa-card__heading" slot="header">{item.title}</h3>
+            <Image
+              slot="image"
+              src={item.heroImage.imageSource.url}
+              alt={item.heroImage.imageSource.title || "Card image"}
+              sources={getSources(item.heroImage.imageSource.url)}
```

Comment:
This fixes the images disappearing at large sizes. The problem was that `getSources` wasn't being told about the image's width so it generated sizes that were larger than the image itself. `sources` is only supposed to be an array when you have a static set of images (like our build-time assets) - if you have a URL generator like `getSources` you can pass that URL generator into the `Image` component as `sources` and it will automatically call it with the image's metadata (currently just height and width) and use that as the sources.

```suggestion
              sources={getSources}
```

Reply:
Thanks for the assist here!